src/gt4py/analysis/passes.py

```diff
@@ -909,14 +913,20 @@ def visit_TernaryOpExpr(self, node: gt_ir.TernaryOpExpr, **kwargs):
         def visit_NativeFuncCall(self, node: gt_ir.NativeFuncCall, **kwargs):
             self.generic_visit(node.args, **kwargs)
 
-            data_type = set(
-                arg.data_type for arg in node.args if arg.data_type != gt_ir.DataType.DEFAULT
-            )
-            if len(data_type) > 1:
-                raise DataTypeSpecificationError(
-                    "Builtin function call with mixed data types", loc=node.loc
-                )
-            data_type = data_type.pop() if len(data_type) else gt_ir.DataType.DEFAULT
+            dtypes_list = [arg.data_type for arg in node.args]
+            dtypes_set = set(dtypes_list) - {gt_ir.DataType.DEFAULT}
```

Comment:
It looks like dtypes_list is not longer needed...
```suggestion
            dtypes_set = set(
                arg.data_type for arg in node.args if arg.data_type != gt_ir.DataType.DEFAULT
            )
```